src/navigator/Navigator.tsx

```diff
@@ -622,83 +607,102 @@ export function MainStackScreen() {
+      options={{ ...mainScreenNavOptions(), ...GoldEducation.navigationOptions }}
     />
     <Navigator.Screen
       name={Screens.AccountKeyEducation}
       component={AccountKeyEducation}
-      options={AccountKeyEducation.navigationOptions}
+      options={{ ...mainScreenNavOptions(), ...AccountKeyEducation.navigationOptions }}
     />
     <Navigator.Screen
       name={Screens.LanguageModal}
       component={Language}
-      options={Language.navigationOptions(true)}
+      options={{ ...mainScreenNavOptions(), ...Language.navigationOptions(true) }}
     />
     <Navigator.Screen
       name={Screens.SelectCountry}
       component={SelectCountry}
-      options={SelectCountry.navigationOptions}
+      options={{ ...mainScreenNavOptions(), ...SelectCountry.navigationOptions }}
     />
     <Navigator.Screen
       name={Screens.SendConfirmationModal}
       component={SendConfirmation}
-      options={sendConfirmationScreenNavOptions}
+      options={{ ...mainScreenNavOptions(), ...sendConfirmationScreenNavOptions }}
     />
     <Navigator.Screen
       name={Screens.SendConfirmationLegacyModal}
       component={SendConfirmationLegacy}
-      options={sendConfirmationLegacyScreenNavOptions}
-    />
-    <Navigator.Screen
-      name={Screens.WalletConnectRequest}
-      component={WalletConnectRequest}
-      options={{
-        ...modalScreenOptions(),
-        ...noHeader,
-        gestureEnabled: false,
-      }}
+      options={{ ...mainScreenNavOptions(), ...sendConfirmationLegacyScreenNavOptions }}
     />
   </>
 )
 
-const mainScreenNavOptions = () => ({
-  ...modalScreenOptions(),
-  headerShown: false,
-})
+function nativeBottomSheets(BottomSheet: typeof RootStack) {
+  return (
+    <>
+      <BottomSheet.Screen
+        name={Screens.WalletConnectRequest}
+        component={WalletConnectRequest}
+        options={{ snapPoints: ['60%'] }}
+      />
+      <BottomSheet.Screen
+        name={Screens.DappKitAccountScreen}
+        component={DappKitAccountScreen}
+        options={{ snapPoints: ['60%'] }}
+      />
+      <BottomSheet.Screen
+        name={Screens.DappKitSignTxScreen}
+        component={DappKitSignTxScreen}
+        options={{ snapPoints: ['60%'] }}
+      />
+    </>
+  )
+}
 
 function RootStackScreen() {
+  const renderBackdrop = React.useCallback(
+    (props: BottomSheetBackdropProps) => (
+      <BottomSheetBackdrop opacity={0.25} appearsOnIndex={0} disappearsOnIndex={-1} {...props} />
+    ),
+    []
+  )
+
   return (
-    <RootStack.Navigator mode="modal">
-      <RootStack.Screen
-        name={Screens.Main}
-        component={MainStackScreen}
-        options={mainScreenNavOptions}
-      />
-      {modalAnimatedScreens(RootStack)}
+    <RootStack.Navigator
+      screenOptions={{ snapPoints: ['100%'], backdropComponent: renderBackdrop }}
+    >
+      <RootStack.Screen name={Screens.Main} component={MainStackScreen} />
+      {nativeBottomSheets(RootStack)}
     </RootStack.Navigator>
```

Comment:
ah this was a reasonable request, unfortunately resulted in a lot of new code to review 😅 

i think it was worth the effort though - i couldn't directly move the `modalAnimatedScreens` back to the root stack because they were using stack styles (as the root stack used to be a stack navigator and now is a bottom sheet navigator). but most of the screens there were just trying to be a full screen modal anyway, so setting the snap point to 100% and hiding the handle component seemed to work the same.

i did do a lot of testing on my iOS simulator and android device, and think that i've not broken anything. the only screens that do have notable changes are the PincodeEnter and SelectCountry screens - those were using back buttons but didn't cover the whole screen, i opted to go for the bottom sheet handle there to make them more like native bottom sheets.